src/getFID.ts

```diff
@@ -59,30 +41,36 @@ export const getFID = (onReport: ReportHandler) => {
   };
 
   const po = observe('first-input', entryHandler as PerformanceEntryHandler);
-  const report = bindReporter(onReport, metric);
+  report = bindReporter(onReport, metric);
 
   if (po) {
     onHidden(() => {
       po.takeRecords().map(entryHandler as PerformanceEntryHandler);
       po.disconnect();
     }, true);
+  }
+
+  if (self.__WEB_VITALS_EXTERNAL_POLYFILL__) {
```

Comment:
Yes, what you've outlined in the bullet points is correct.

> I have to ask: if folks have the web-vitals.js library loaded, and it now already comes with the FID polyfill, should we support recording always FID+BFCache restores? Or, is web-vitals.js expected to be loaded too late to capture initial FID reliably?

The latter, it could be loaded too late, which is why using the polyfill requires adding a snippet of code to the `<head>`. The polyfill does work in the bfcache restore case because the web-vitals bundle is already loaded at that point.

Also, since may not be obvious, the `self.__WEB_VITALS_EXTERNAL_POLYFILL__` conditional blocks get stripped out at build time. They're used to build two versions of the the `getFID()` module.